drivers/filesystems/ext2/src/ext4/ext4_extents.c

```diff
@@ -83,10 +87,16 @@ static inline void ext_debug(char *str, ...)
 {
 }
 #if TRUE
+#ifdef __REACTOS__
 #define EXT4_ERROR_INODE(inode, str, ...) do {                      \
             DbgPrint("inode[%p]: " str "\n", inode, ##__VA_ARGS__); \
         } while(0)
 #else
+#define EXT4_ERROR_INODE(inode, str, ...) do {                      \
```

Comment:
Isn't that definition more correct than the ifdef'ed ReactOS one (minus the va_args that better need the ## in front of it)?
(or, is it possible to find why the ROS one removed the token pasting ## operator in front of str?)

Reply:
I'm not familiar enough (yet) with this part of ext2fsd to answer this question.
all I can see is that it has been part of the integration of ext2fsd in our codebase https://github.com/reactos/reactos/commit/897634ce5bbb47b0f7c4ec48200baa77eb30b4f4 .
And I can trace back its origin to a patch in the related jira ticket https://jira.reactos.org/browse/CORE-10272 
https://jira.reactos.org/secure/attachment/31814/ext2fsd.patch  
but in the different conversation I haven't found an explanation about why they did that change

Reply:
@HBelusca we cannot put ##str as str is always just a single string, one token, so no concatenation of token is possible here

Reply:
yes actually you're correct. I've looked at the usages of `EXT4_ERROR_INODE` and it's used like:
```c
EXT4_ERROR_INODE(inode, "Some string with %s formatting", "something");
```
and so indeed, we cannot use the stringizer operator on `str`, otherwise it would re-stringize the string ^^